google/cloud/pubsub_v1/subscriber/_protocol/dispatcher.py

```diff
@@ -120,17 +121,38 @@ def dispatch_callback(self, items: Sequence[RequestItem]) -> None:
         _LOGGER.debug("Handling %d batched requests", len(items))
 
         if batched_commands[requests.LeaseRequest]:
-            self.lease(batched_commands.pop(requests.LeaseRequest))
+            lease_requests = cast(
+                List[requests.LeaseRequest],
+                batched_commands.pop(requests.LeaseRequest),
+            )
+            self.lease(lease_requests)
```

Comment:
The typing contortions here are a sign that using a single `batched_commands` mapping is a broken design.  It would be far cleaner to keep four separate lists.

Reply:
Fair point, will refactor.